flake.nix

```diff
@@ -0,0 +1,125 @@
+    inherit (nixpkgs) lib;
+  in
+    (dream2nix.lib.makeFlakeOutputs {
+      systems = ["x86_64-linux"];
+      config.projectRoot = ./.;
+      config.packagesDir = "./locks";
+      source = ./.;
+      settings = [
+        {subsystemInfo.noDev = false;}
+      ];
+      packageOverrides = let
+        system = "x86_64-linux";
+        toplevel = "cardano-sdk";
+        subpackages = lib.filterAttrs (k: v: lib.hasPrefix "@cardano-sdk" k) self.packages.${system};
+        subpackagesNames = builtins.attrNames subpackages;
+        topDependencies =
+          builtins.removeAttrs
+          self.packages.${system}.${toplevel}.dependencies
+          (subpackagesNames ++ [toplevel]);
+      in
+        lib.recursiveUpdate (
+          builtins.mapAttrs (n: _: let
+            otherNames = lib.remove n subpackagesNames;
+            otherSubpackages = builtins.removeAttrs subpackages [n];
+          in {
+            # install-method.installMethod = "copy";
+            add-inputs.buildInputs = old: old ++ (lib.attrValues topDependencies) ++ [pkgs.nodejs-14_x.pkgs.yarn];
+            install-symlinks.postInstall = "ln -s $out/lib/node_modules/${n}/dist $out/dist";
+            add-type-notation.prePatch = lib.optionalString (n == "@cardano-sdk/core")
+            ''
+              substituteInPlace ./src/Cardano/util/subtractValueQuantities.ts \
+                --replace "../types"  "@cardano-sdk/core/src/Cardano/types"
+            '';
+            fix-relative-references.postPatch = let
+              references = {
+                "@cardano-sdk/blockfrost" = [ "core" "util" ];
+                "@cardano-sdk/cardano-services-client" = [ "core" "util" ];
+                "@cardano-sdk/cardano-services" = [ "core" "ogmios" "rabbitmq" "util" ];
+                "@cardano-sdk/cip2" = [ "core" "util" ];
+                "@cardano-sdk/cip30" = [ "core" ];
+                "@cardano-sdk/core" = [ "util" ];
+                "@cardano-sdk/e2e" = [ "util" "util-dev" "wallet" "cardano-services" ];
+                "@cardano-sdk/golden-test-generator" = [ "core" "util" ];
+                "@cardano-sdk/ogmios" = [ "core" ];
+                "@cardano-sdk/rabbitmq" = [ "core" "util" ];
+                "@cardano-sdk/util" = [];
+                "@cardano-sdk/util-dev" = [ "core" ];
+                "@cardano-sdk/util-rxjs" = [];
+                "@cardano-sdk/wallet" = [ "cip2" "cip30" "core" "util-rxjs" "util"];
+                "@cardano-sdk/web-extension" = [ "cip30" "core" "wallet" "util-rxjs" "util"];
+              };
+              replacer = lib.concatMapStrings (name: ''
+                --replace \
+                  '"path": "../../${name}/src"' \
+                  '"path": "../node_modules/@cardano-sdk/${name}/src"' \
+              '');
+            in ''
+              cp ${./tsconfig.json} ./src/tsconfig-root.json
+              mkdir build
+              cp ${./build/cjs-package.json} ./build/cjs-package.json
```

Comment:
Can we change package build script to from `build` to `build:cjs`? `build` is making 2 builds (into different module formats) and we only need 1 here.

Also, `cjs-package.json` and `esm-package.json` are not going to be needed then so we could simplify the nix file.

Reply:
If these aren't going to be needed, I'd prefer them to be removed from the `build` entrypoint in `packages.json`. Packages json is our upstream here and any semantic deviation probably would require more (to be maintained) explanatory prose than we would gain by excluding this case.

If we remove this, anybody will immediately ask "why is this semantically different for no obvious reason"?

The second build never has made any problems, after all.

Reply:
In context of SDK, `build` script prepares packages to be published to `npm` and we do want to support multiple module formats. Of course we could have a separate script for that, but it boils down to what is the primary purpose for these packages